library/core/src/ptr/mod.rs

```diff
@@ -16,12 +16,16 @@
 //! provided at this point are very minimal:
 //!
 //! * A [null] pointer is *never* valid, not even for accesses of [size zero][zst].
-//! * All pointers (except for the null pointer) are valid for all operations of
-//!   [size zero][zst].
 //! * For a pointer to be valid, it is necessary, but not always sufficient, that the pointer
 //!   be *dereferenceable*: the memory range of the given size starting at the pointer must all be
 //!   within the bounds of a single allocated object. Note that in Rust,
 //!   every (stack-allocated) variable is considered a separate allocated object.
+//! * Even for operations of [size zero][zst], the pointer must not be "dangling" in the sense of
+//!   pointing to deallocated memory. However, casting any non-zero integer *literal* to a pointer is
+//!   valid for zero-sized accesses. This corresponds to writing your own allocator; allocating
+//!   zero-sized objects is not very hard. In contrast, when you use the standard allocator, after
+//!   memory got deallocated, even zero-sized accesses to that memory are invalid.
+//!   The canonical way to obtain a pointer that is valid for zero-sized accesses is [`NonNull::dangling`].
```

Comment:
This is rather unfortunate naming: the text says "the pointer must not be dangling" and then proceeds to recommend calling `NonNull::dangling`...

Is there another term we could use instead of the first "dangling"?

Reply:
```diff
- the pointer must not be "dangling" in the sense of pointing to deallocated memory
+ the pointer must not be pointing to deallocated memory
```

maybe?

Reply:
I tried something like this, WDYT?

Reply:
It's great!👌